src/purejavahidapi/windows/HidDevice.java

```diff
@@ -199,12 +197,21 @@ synchronized public int setOutputReport(byte reportID, byte[] data, int length)
 				}
 			}
 
+			Log("setOutputReport1");
+			if (WAIT_OBJECT_0 != WaitForSingleObject(m_OutputReportOverlapped.hEvent, 1000)) {
```

Comment:
Since the "wait" is now happening here, the `bWait` parameter of `GetOverlappedResult` should be changed to false.  That way it's clear to anyone maintaining the code that `GetOverlappedResult` can't block indefinitely.

Reply:
I tried that, and it fails. So I kept it set to 'true'. Let me know if you want me to investigate.

Reply:
Interesting... what was the `GetLastError`?

Reply:
IO_INCOMPLETE: Overlapped I/O event is not in a signaled state

Reply:
After `WaitForSingleObject` returns (without a timeout error), the event should be in the signaled state.  Was it somehow set to auto-reset?  Maybe something is happening in the JNA layer that we don't expect.

Reply:
Right. I did have to create it with manual reset option before it worked at all (even added a comment to relate its importance). So JNA layer indeed is a suspect, but it worked with 'true' flag so I left it with it.